languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/pt-BR/grammar.xml

```diff
@@ -169,17 +169,114 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+            <example>O treinamento do agente foi árduo.</example>
+            <example>Ele era um agente da KGB.</example>
+            <example>Esse agente não teve uma boa atuação.</example>
+        </antipattern>
+        <antipattern>
+            <token>agente</token>
+            <token postag_regexp="yes" postag="R." min="0" max="-1"/>
+            <token postag_regexp="yes" postag="A.+|SP.+(:DA.+)|VMP.+"/>
+            <example>Um filme com agente secreto.</example>
+            <example>Ela era uma gente muito bem treinada.</example>
+            <example>Seu filho sempre foi um agente do caos.</example>
+        </antipattern>
+        <antipattern>
+            <token postag_regexp="yes" postag="A.+|SP.+(:DA.+)"/>
+            <token>agente</token>
+            <example>Ele sempre foi um excelente agente.</example>
+        </antipattern>
+        <!-- TODO improve disambiguator for basic ordinals -->
+        <antipattern>
+            <token regexp="yes">primeir[oa]|segund[oa]|melhor</token>
+            <token>agente</token>
+            <example>Sua primeira agente foi a poderosa.</example>
+        </antipattern>
+        <antipattern>
+            <token>agente</token>
+            <!-- TODO turn this into an entity, it could yet prove useful -->
+            <token regexp="yes">\p{L}+([aei](nte|dor)|[âêô]nic[oa]|os[oa]|lógic[oa]|óide)</token>
+            <!-- TODO add to dictionary, but let's keep this here as
+            this derivation is relatively productive. -->
+            <example>Agente vitrificante.</example>
+            <example>Agente descolorizante.</example>
+            <example>Agente umidificador.</example>
+            <example>Agente infeccioso.</example>
+            <example>Agente etiológico.</example>
+            <example>Agente mutagênico.</example>
+        </antipattern>
+        <antipattern>
+            <token>agente</token>
+            <token spacebefore="no" min="0">:</token>
+            <token regexp="yes">\p{Lu}[\p{L}\d]*|\d+[\p{L}\d]*</token>
+            <example>Agente Jack Bauer.</example>
+            <!-- <example>Agente K.D. Lang.</example> -->
+            <example>Agente 007.</example>
+        </antipattern>
+        <antipattern>
+            <token>agente</token>
+            <token spacebefore="no">/</token>
+            <token postag_regexp="yes" postag="N.+"/>
+            <example>Agente/diretor Smith.</example>
+        </antipattern>
+        <antipattern>
+            <token regexp="yes">\p{Lu}\p{L}+</token>
+            <token spacebefore="no" min="0">,</token>
+            <token min="0">como</token>
+            <token>agente</token>
+            <example>Francisco Pestana, como Agente.</example>
+            <example>Frank Hubbard como agente número três.</example>
+            <example>Hank McGlaurie, agente e parceiro de Mala.</example>
+        </antipattern>
+        <antipattern>
+            <!-- TODO create/find entity with productive prefixes -->
```

Comment:
`<!ENTITY prefixos "infanto|luso|franco|ítalo|al|hemi|mono|di|tri|tetra|penta|hexa|hepta|octa|enea|deca|bi|semi|e|em|en|endo|ec|ecto|ex|exo|epi|dia|meso|para|peri|pros|a|an|ana|anfi|anti|apo|arca|arce|arqui|dis|des|de|diplo|eu|ev|hiper|hipo|macro|mega|meta|micro|oligo|poli|sim|sin|sis|so|sob|sub|sobre|super|supra|tra|tras|trans|tres|cis|ultra|vice|vis|ab|ad|ambi|ante|con|co|i|im|in|ob|oc|quadri|pen|per|pro|pró|pós|contra|abs|bene|bem|aquém|bis|extra|intra|justa|pre|pré|preter|re|retro|aero|agro|antropo|idio|megalo|metra|necro|nefro|odonto|oftalmo|onto|pneumo|bio|cardio|fito|fisis|gero|gimno|hialo|hiero|holo|leuco|mero|neo|quilo|quiro|rino|rizo|xeno|xilo|zoo|aristo|arque[oó]|anto|atmo|auto|baro|caco|cali|a?c[eé]falo|cito|copro|cosmo|crono|dico|eno|entero|etno|filia|flama|fono|gastro|hemo|hidro|higro">`